src/core/handle.ml

```diff
@@ -187,6 +197,35 @@ let handle_cmd : sig_state -> p_command -> sig_state * proof_data option =
       let d = if op then None else Some(t) in
       let ss = Sig_state.add_symbol ss e Defin x a impl d in
       (ss, None)
+  | P_inductive(e, s, t, tl)     ->
+      (* Obtaining of the global type *)
+      let type_symbol_head = match t with
+        | None   -> none P_Type
+        | Some a -> a
+      in
+      (* Add the head symbol in the signature *)
+      let ss = handle_symbol ss e Defin s [] type_symbol_head in
+      let type_symbol_head = scope_term e ss Env.empty type_symbol_head in
+      (* Create the body of the inductive type *)
+      let rec add_constructor : sig_state -> sym list -> (ident * p_term) list
+                                -> sig_state * sym list = fun ss_cur syml l ->
+        match l with
+        | []                -> (ss_cur, syml)
+        | (ident, pterm)::q ->
+            let ss = handle_symbol ss_cur e Injec ident [] pterm in
+            let term = scope_term e ss Env.empty pterm in (*empty env ?*)
+            let new_sym = create_sym (ident.elt) term Injec e in
+            add_constructor ss ((new_sym)::syml) q
```

Comment:
create_sym is useless and wrong here as the right symbol is created in handle_symbol. You can get it back by changing the code of Sig_state.add_symbol and handle_symbol so that they return a sig_state * sym.
Then, you could write something like:
```
let (ss,typ) = handle_symbol ss e Const s [] t in
let rec add_cons : sig_state * sym list -> ident * p_term -> sig_state * sym list
= fun (ss, cons) (id, a) ->
  let (ss, c) = handle_symbol ss e Const id [] a in
  (ss, c::cons)
in
let (ss, cons) = List.fold_left add_cons (ss, []) tl in
let pr = Inductive.principle typ cons in
Sign.add_inductive ss.signature typ cons pr;
(ss, None)
```

Reply:
Why is it wrong? The function `create_sym` is only used to create my record.
And it is more general than `new_sym`..
I can change the signature, but this change a lot of things in the code, no? (just for my function)

Reply:
It's wrong because handle_symbol already creates a symbol, and create_sym creates a new one as well which is different. That's why I say that it is sufficient that handle_symbol returns the symbol that it created. Thanks Amélie to read my comments carefully and try to take them into account.

Reply:
Done